lib/matplotlib/axis.py

```diff
@@ -124,16 +124,32 @@ def __init__(
                 zorder = mlines.Line2D.zorder
         self._zorder = zorder
 
-        grid_color = mpl._val_or_rc(grid_color, "grid.color")
-        grid_linestyle = mpl._val_or_rc(grid_linestyle, "grid.linestyle")
-        grid_linewidth = mpl._val_or_rc(grid_linewidth, "grid.linewidth")
+        grid_color = (
+            mpl._val_or_rc(grid_color, "grid.color")
+            if mpl.rcParams[f"grid.{major_minor}.color"] == "none"
```

Comment:
```suggestion
            if mpl.rcParams[f"grid.{major_minor}.color"] is None
```
see comment on validation